File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/secrets/SecretsApi.java
package co.uk.swarmbit.docker.api.secrets;

import co.uk.swarmbit.docker.api.common.json.SecretCreateResponseJson;
import co.uk.swarmbit.docker.api.common.json.SecretJson;
import co.uk.swarmbit.docker.api.secrets.parameters.SecretCreateParameters;
import co.uk.swarmbit.docker.api.secrets.parameters.SecretUpdateParameters;
import co.uk.swarmbit.docker.api.secrets.parameters.SecretsListParameters;

import java.util.List;

public interface SecretsApi {

    List<SecretJson> listSecrets(String swarmId, SecretsListParameters parameters);

    SecretJson inspectSecret(String swarmId, String id);

    SecretCreateResponseJson createSecret(String swarmId, SecretCreateParameters parameters);

    void deleteSecret(String swarmId, String id);

    void updateSecret(String swarmId, String id, SecretUpdateParameters parameters);
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/secrets/SecretsApi.kt
package com.swarmbit.docker.api.secrets

import com.swarmbit.docker.api.secrets.model.Secret
import com.swarmbit.docker.api.secrets.model.SecretCreateResponse
import com.swarmbit.docker.api.secrets.parameters.SecretsCreateParameters
import com.swarmbit.docker.api.secrets.parameters.SecretsListParameters
import com.swarmbit.docker.api.secrets.parameters.SecretsUpdateParameters

interface SecretsApi {
    fun listSecrets(swarmId: String, parameters: SecretsListParameters): List<Secret>
    fun inspectSecret(swarmId: String, id: String): Secret
    fun createSecret(swarmId: String, parameters: SecretsCreateParameters): SecretCreateResponse
    fun deleteSecret(swarmId: String, id: String)
    fun updateSecret(swarmId: String, id: String, parameters: SecretsUpdateParameters)
}
